pylint/checkers/similar.py

```diff
@@ -798,33 +798,18 @@ def set_option(self, optname, value, action=None, optdict=None):
         Overridden to report options setting to Similar
         """
         # pylint: disable-next=fixme
-        # TODO: Refactor after OptionProvider has been moved to argparse
-        BaseChecker.set_option(self, optname, value, action, optdict)
+        # TODO: Deprecate the non required arguments
+        self.linter.set_option(optname, value)
         if optname == "min-similarity-lines":
-            self.min_lines = (
-                getattr(self.linter.namespace, "min_similarity_lines", None)
-                or self.config.min_similarity_lines
-            )
+            self.min_lines = self.linter.namespace.min_similarity_lines
         elif optname == "ignore-comments":
```

Comment:
Do we need the conditional and set_option here ? Can't we do 
```python
self.min_lines = self.linter.namespace.min_similarity_lines
...
self.ignore_signatures = self.linter.namespace.ignore_signatures
```
Directly inside the constructor of ``Similar`` ?

Reply:
We probably can, but I need to refactor this anyway. (See the TODO).

I'll take a look at these assignments at the same time. I'm likely going to deprecate this complete method and just use `set_option` from the `Manager`. No need to duplicate if both work fine.